get-fedora-latest-config

```diff
@@ -1,64 +1,83 @@
 #!/bin/bash
+# vim: set ts=4 sw=4 sts=4 et :
 
 set -e
 if [ "${VERBOSE:-0}" -ge 2 ] || [ "${DEBUG:-0}" -eq 1 ]; then
+    debug=1
     set -x
 fi
 
 localdir="$(dirname "$(readlink -f "$0")")"
 releasever="$1"
-testing="$2"
 
 kernelver="$(cat "$localdir/version")"
 kernelsrc="linux-$kernelver"
 
-key="$localdir/keys/RPM-GPG-KEY-fedora-$releasever-primary"
+errecho() {
+    >&2 echo "$@"
+}
 
-[[ "x$releasever" == "x" ]] && { echo "Please provide Fedora release version as first argument, e.g. '29'."; exit 1; }
-
-# baseurl for latest kernel rpm
-if [ "$testing" == "1" ]; then
-    url="https://dl.fedoraproject.org/pub/fedora/linux/development/$releasever/Everything/x86_64/os/Packages/k"
-else
-    url="https://dl.fedoraproject.org/pub/fedora/linux/updates/$releasever/Everything/x86_64/Packages/k"
+# example of releasever: '29' or 'rawhide'
+if [ "x$releasever" != "x" ]; then
+    if [[ ! "$releasever" =~ ^[1-9][0-9]$ ]] && [ "$releasever" != "rawhide" ]; then
+        errecho "Invalid release format"
+        exit 1
+    fi
+elif [ "x$releasever" == "x" ]; then
+    listver="$(curl -s -L https://dl.fedoraproject.org/pub/fedora/linux/releases 2> /dev/null)"
+    releasever="$(echo "$listver" | sed -e 's/<[^>]*>//g' | awk '{print $1}' | grep -o "[1-9][0-9]" | tail -1)"
+    if ! [[ "$releasever" =~ ^[1-9][0-9]$ ]]; then
+        errecho "An error occurred while trying to determine latest Fedora version"
+        exit 1
+    fi
 fi
 
 # get the latest kernel rpm
-content="$(curl "$url/")"
-latestrpm="$(echo "$content" | sed -e 's/<[^>]*>//g' | awk '{print $1}' | grep -E "kernel-core-[0-9]+.[0-9]+.[0-9]+-[0-9]+.fc$releasever.x86_64.rpm")"
-latestver="$(echo "$latestrpm" | sed 's/kernel-core-//; s/\.rpm//')"
+latestver=$(dnf -q repoquery kernel-core --disablerepo=*modular* --releasever="$releasever" | tail -1 | cut -d ':' -f2)
```

Comment:
I'd also disable qubes repositories, otherwise it will fail for not yet uploaded versions (and rawhide).

Reply:
I enabled only Fedora repos.